lib/puppet/defaults.rb

```diff
@@ -1862,6 +1862,19 @@ def self.initialize_default_settings!(settings)
       certificate request to be signed. A value of `unlimited` will cause puppet agent
       to ask for a signed certificate indefinitely.
       #{AS_DURATION}",
+    },
+    :waitforlock => {
+      :default  => false,
+      :type     => :boolean,
+      :desc     => "Whether to wait for already running puppet agent to finish before starting.",
+    },
+    :maxwaitforlock => {
+      :default  => "1m",
+      :type     => :ttl,
+      :desc     => "The maximum amount of time the Puppet agent should wait for
+      already running puppet agent to finish before starting. A value of `unlimited`
+      will cause puppet agent to ask for a signed certificate indefinitely.
```

Comment:
looks like a copy/paste error, maybe "will cause puppet agent to wait indefinitely."?